### PR TITLE
linuxPackages.lttng-modules: 2.10.5 -> 2.12.6

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](./CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/lttng-modules/default.nix [hunk 1]
```diff
@@ -28,11 +27,8 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Linux kernel modules for LTTng tracing";
     homepage = "https://lttng.org/";
-    license = with licenses; [ lgpl21 gpl2 mit ];
+    license = with licenses; [ lgpl21Only gpl2Only mit ];
     platforms = platforms.linux;
     maintainers = [ maintainers.bjornfor ];
-    broken = builtins.compareVersions kernel.version "3.18" == -1
-      || builtins.compareVersions kernel.version "4.16" == 1;
```
Comment (fortuneteller2k) [inline pkgs/os-specific/linux/lttng-modules/default.nix:35]: Temporary, until I get the `nixpkgs-review` result.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 125797` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.lttng-modules</li>
<li>linuxPackages_hardkernel_4_14.lttng-modules</li>
<li>linuxPackages_hardkernel_latest.lttng-modules</li>
<li>linuxPackages_latest-libre.lttng-modules</li>
</ul>
</details>
<details>

  <summary>12 packages failed to build and are new build failures:</summary>
  <ul>
<li>linuxPackages.lttng-modules: <a href="https://termbin.com/rgik7">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/e7612636f5f1df5e2c49dca03283f5e6/raw/index.html">pretty log </a></li>
<li>(linuxPackages_5_10.lttng-modules): log was empty</li>
<li>linuxPackages_5_11.lttng-modules: log was empty</li>
<li>linuxPackages_5_12.lttng-modules: log was empty</li>
<li>(linuxPackages_latest.lttng-modules): log was empty</li>
<li>linuxPackages_5_4.lttng-modules: log was empty</li>
<li>linuxPackages_hardened.lttng-modules: log was empty</li>
<li>linuxPackages_latest_hardened.lttng-modules: log was empty</li>
<li>linuxPackages_lqx.lttng-modules: log was empty</li>
<li>linuxPackages_testing_bcachefs.lttng-modules: log was empty</li>
<li>linuxPackages_xanmod.lttng-modules: log was empty</li>
<li>linuxPackages_zen.lttng-modules: log was empty</li>
  </ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>linuxPackages_4_14.lttng-modules</li>
<li>linuxPackages_4_19.lttng-modules</li>
<li>linuxPackages_4_4.lttng-modules</li>
<li>linuxPackages_4_9.lttng-modules</li>
</ul>
</details>

**Comment by fortuneteller2k** [comment]
Bah, didn't see that there was a `2.12.6` version, which builds just fine on newer kernels.

```
  INSTALL /build/lttng-modules-2.12.6/tests/lttng-test.ko
  DEPMOD  5.12.7
Warning: modules_install: missing 'System.map' file. Skipping depmod.
make[1]: Leaving directory '/nix/store/qnwxscgnzk2yhdh3ikhavza8k1b2vacd-linux-5.12.7-dev/lib/modules/5.12.7/build'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/9qh0w8v2m5qkqca6sc6qzygar3ydng6j-lttng-modules-5.12.7-2.12.6
strip is /nix/store/77i6h1kjpdww9zzpvkmgyym2mz65yff1-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/9qh0w8v2m5qkqca6sc6qzygar3ydng6j-lttng-modules-5.12.7-2.12.6/lib
patching script interpreter paths in /nix/store/9qh0w8v2m5qkqca6sc6qzygar3ydng6j-lttng-modules-5.12.7-2.12.6
checking for references to /build/ in /nix/store/9qh0w8v2m5qkqca6sc6qzygar3ydng6j-lttng-modules-5.12.7-2.12.6...
/nix/store/9qh0w8v2m5qkqca6sc6qzygar3ydng6j-lttng-modules-5.12.7-2.12.6
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 125797` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.lttng-modules</li>
<li>linuxPackages_latest-libre.lttng-modules</li>
</ul>
</details>
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>linuxPackages.lttng-modules</li>
<li>linuxPackages_4_14.lttng-modules</li>
<li>linuxPackages_4_19.lttng-modules</li>
<li>linuxPackages_4_4.lttng-modules</li>
<li>linuxPackages_4_9.lttng-modules</li>
<li>linuxPackages_5_11.lttng-modules</li>
<li>linuxPackages_5_12.lttng-modules</li>
<li>linuxPackages_5_4.lttng-modules</li>
<li>linuxPackages_hardened.lttng-modules</li>
<li>linuxPackages_latest_hardened.lttng-modules</li>
<li>linuxPackages_lqx.lttng-modules</li>
<li>linuxPackages_testing_bcachefs.lttng-modules</li>
<li>linuxPackages_xanmod.lttng-modules</li>
<li>linuxPackages_zen.lttng-modules</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125797` at 1f055097 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_hardkernel_4_14.lttng-modules</li>
    <li>linuxPackages_hardkernel_latest.lttng-modules</li>
    <li>linuxPackages_testing_bcachefs.lttng-modules</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/17c0806e8cd31009a32d0eac60ae82f2">linuxPackages_4_4.lttng-modules</a></li>
    <li><a href="https://gist.github.com/228082c3bc5b5132e7090168aacc4e20">linuxPackages_xanmod.lttng-modules</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>linuxPackages-libre.lttng-modules</li>
    <li>linuxPackages_4_9.lttng-modules</li>
  </ul>
</details>
<details>
  <summary>11 packages built successfully:</summary>
  <ul>
    <li>linuxPackages.lttng-modules (linuxPackages_5_10.lttng-modules)</li>
    <li>linuxPackages_4_14.lttng-modules</li>
    <li>linuxPackages_4_19.lttng-modules</li>
    <li>linuxPackages_5_11.lttng-modules</li>
    <li>linuxPackages_5_12.lttng-modules (linuxPackages_latest.lttng-modules)</li>
    <li>linuxPackages_5_4.lttng-modules</li>
    <li>linuxPackages_hardened.lttng-modules</li>
    <li>linuxPackages_latest-libre.lttng-modules</li>
    <li>linuxPackages_latest_hardened.lttng-modules</li>
    <li>linuxPackages_lqx.lttng-modules</li>
    <li>linuxPackages_zen.lttng-modules</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 125797 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages marked as broken and skipped:</summary>  <ul>    <li>linuxPackages-libre.lttng-modules</li>    <li>linuxPackages_5_11.lttng-modules</li>    <li>linuxPackages_hardkernel_4_14.lttng-modules</li>    <li>linuxPackages_hardkernel_latest.lttng-modules</li>    <li>linuxPackages_latest-libre.lttng-modules</li>  </ul></details><details>  <summary>13 packages built:</summary>  <ul>    <li>linuxPackages.lttng-modules (linuxPackages_5_10.lttng-modules)</li>    <li>linuxPackages_4_14.lttng-modules</li>    <li>linuxPackages_4_19.lttng-modules</li>    <li>linuxPackages_4_4.lttng-modules</li>    <li>linuxPackages_4_9.lttng-modules</li>    <li>linuxPackages_5_12.lttng-modules (linuxPackages_latest.lttng-modules)</li>    <li>linuxPackages_5_4.lttng-modules</li>    <li>linuxPackages_hardened.lttng-modules</li>    <li>linuxPackages_latest_hardened.lttng-modules</li>    <li>linuxPackages_lqx.lttng-modules</li>    <li>linuxPackages_testing_bcachefs.lttng-modules</li>    <li>linuxPackages_xanmod.lttng-modules</li>    <li>linuxPackages_zen.lttng-modules</li>  </ul></details>

**Pull request merged**